scipy/stats/_stats_py.py

```diff
@@ -9531,3 +9533,114 @@ def rankdata(a, method='average', *, axis=None, nan_policy='propagate'):
         result[nan_indexes] = np.nan
 
     return result
+
+
+def expectile(a, alpha=0.5, *, weights=None):
+    r"""Compute the expectile.
+
+    Expectiles are a generalization of the expectation in the same way as
+    quantiles are a generalization of the median. The expectile at level
+    `alpha = 0.5` is the mean (average). See Notes for more details.
+
+    Parameters
+    ----------
+    a : array_like
+        Input array or object that can be converted to an array.
+    alpha : float, default=0.5
+        The level of the expectile; `alpha=0.5` gives the mean.
+    weights : array_like, optional
+        The sample or case `weights` array must be broadcastable to the same
+        shape as `a`. Default is None, which gives each value a weight of 1.0.
+        An integer valued weight element acts like repeating the corresponding
+        observation in `a` that many times. See Notes for more details.
+
+    Returns
+    -------
+    expectile : ndarray
+        The empirical expectile at level `alpha`.
+
+    See Also
+    --------
+    numpy.mean : Arithmetic average
+    numpy.quantile : Quantile
+
+    Notes
+    -----
+    In general, the expectile at level `alpha` (:math:`\alpha`) of a
+    probability distribution :math:`F` with :math:`X\sim F` is given by the
+    unique solution :math:`t` of:
```

Comment:
I misread this originally, and I think this would have helped to be a bit more explicit.
```suggestion
    cumulative distribution function :math:`F` with :math:`X\sim F`
    is given by the unique solution :math:`t` of:
```
Just "distribution function" would also probably work for me, but not "probability distribution" without "function".

Also, I've not seen a distribution function on the right side of $\sim$ - only another random variable. Perhaps that is fine; my formal stats background is not strong.

I might have described this as 
>  The expectile at level $\alpha$ of a random variable $X$ with cumulative distribution function $F$ is the unique solution...

But since $F$ doesn't appear in the equation, it seems unnecessary to define it until we get down to the equation with the empirical distribution function. I think it would help to add:

> Equivalently, this can be written in terms of the distribution function as....

and then give the equation that uses integrals over $F$ instead of $E(X)$. I think that would provide a missing link between the random variable definition to the sample definition for noobs like me. Then the chain of reasoning would be

Definition in terms of random variables -> *definition in terms of cumulative distribution function* -> definition in terms of empirical distribution function -> definition in terms of sample

Reply:
Hihi. $X\sim F$, e.g. $X\sim N(\mu,\sigma)$, is the most basic used of $\sim$:smirk:

Reply:
Sure, that's all over the place, but doesn't $N$ refer to the random variable itself, not its CDF?

Reply:
> Sure, that's all over the place, but doesn't 
> N
>  refer to the random variable itself, not its CDF?

You’re mostly right. Typically the left hand side is a random variable and the right hand side is a probability distribution (not a random variable) and it would be read as “$X$ is distributed as”. It’s an abuse of notation to write $X \sim F$ where $X$ is a random variable and $F$ is a CDF and I think we should avoid this in the docs.

Reply:
I guess $N$ can mean both, rv or a distribution function.
I'm really *not* the expert on stochastics/probability theory. Klenke "Probability Theory" uses $X\sim F$ with distribution function $F$, but that is more or less equivalent to $X\sim P$ for probability measure $P$. And therefore, it also makes sense for 2 rvs to state $X\sim Y$ (but not used in Klenke). Interestingly, this notation is not used in wikipedia on the important pages :shrug:

Reply:
@steppi You were a bit faster than me. Now one could get the impression that I "would like to know better than you". I happy to change the tilde into words.

Reply:
> @steppi You were a bit faster than me. Now one could get the impression that I "would like to know better than you". I happy to change the tilde into words.

No worries. I think it’s Ok for textbooks to use non-standard notation as long as it’s defined and used consistently. In grad school I had some professors use notation like yours in their lectures, but they explicitly pointed out that it was an abuse. For our docs I think we should go with something more standard though.

Reply:
Fixed in https://github.com/scipy/scipy/pull/17039/commits/3fd19af872f28b87c6d175afb2f7c3998c2cb848.